pkg/agent/openflow/pipeline.go

```diff
@@ -181,55 +183,49 @@ func (c *client) podClassifierFlow(podOFPort uint32) binding.Flow {
 }
 
 // connectionTrackFlows generates flows that redirect traffic to ct_zone and handle traffic according to ct_state:
-// 1) commit new connections to ct that sent from non-gateway.
-// 2) Add ct_mark on traffic replied from the host gateway.
-// 3) Cache src MAC if traffic comes from the host gateway and rewrite the dst MAC on traffic replied from Pod to the
-// cached MAC.
-// 4) Drop all invalid traffic.
+// 1) commit new connections to ct if the packet is sent from non-gateway.
```

Comment:
correct me if I'm wrong, but I think we commit *all* connections, not just the ones which come from the gateway. We have this flow:
```
table=31, priority=190,ct_state=+new+trk,ip actions=ct(commit,table=40,zone=65520)
```
and we use that for the policy tables

Reply:
Actually we commit all connections to ct. But the flow entry you mentioned will commit non-gateway connections. As for the connections from gateway, it should hit another flow `table=31, priority=200,ct_state=+new+trk,ip,reg0=0x1/0xffff actions=ct(commit,table=40,zone=65520,exec(load:0x20->NXM_NX_CT_MARK[])`
The second flow has a higher priority, so the packets sent from gateway interface would hit it firstly, and other non-gateway traffic would hit the flow you mentioned.

Reply:
I will update the comments and remove the "if" conditions.